core/backend/src/rpc-impl/RpcBriefcaseUtility.ts

```diff
@@ -21,39 +25,110 @@ const loggerCategory: string = BackendLoggerCategory.IModelDb;
+   * Download and open a checkpoint or briefcase, ensuring the operation completes within a default timeout. If the time to open exceeds the timeout period,
    * a RpcPendingResponse exception is thrown
    * @param requestContext
    * @param tokenProps
    * @param syncMode
    */
-  public static async openWithTimeout(requestContext: AuthorizedClientRequestContext, tokenProps: IModelRpcProps, syncMode: SyncMode): Promise<IModelConnectionProps> {
-    requestContext.enter();
-    Logger.logTrace(loggerCategory, "RpcBriefcaseUtility.openWithTimeout: Received open request", () => ({ ...tokenProps, syncMode }));
-
-    /*
-     * Download the briefcase
-     * Note: This sets up a race between the specified timeout period and the open. Throws a RpcPendingResponse exception if the
-     * timeout happens first.
-     */
-    const timeout = 1000; // 1 second
-    const { contextId, iModelId, changeSetId } = tokenProps;
-    const briefcaseProps: BriefcaseProps | void = await BeDuration.race(timeout, BriefcaseManager.download(requestContext, contextId!, iModelId!, { syncMode }, IModelVersion.asOfChangeSet(changeSetId!)));
+  public static async open(requestContext: AuthorizedClientRequestContext, tokenProps: IModelRpcProps, syncMode: SyncMode, timeout: number = 1000): Promise<IModelDb> {
     requestContext.enter();
+    Logger.logTrace(loggerCategory, "RpcBriefcaseUtility.open", () => ({ ...tokenProps, syncMode }));
+
+    if (syncMode === SyncMode.PullOnly || syncMode === SyncMode.PullAndPush) {
+      const briefcaseDb = await BeDuration.race(timeout, this.openBriefcase(requestContext, tokenProps, syncMode));
+      requestContext.enter();
 
-    if (briefcaseProps === undefined) {
-      Logger.logTrace(loggerCategory, "RpcBriefcaseUtility.openWithTimeout: Issued pending status", () => ({ ...tokenProps, syncMode }));
-      throw new RpcPendingResponse();
+      if (briefcaseDb === undefined) {
+        Logger.logTrace(loggerCategory, "Open briefcase - pending", () => ({ ...tokenProps }));
+        throw new RpcPendingResponse();
+      }
+      // note: usage is logged in BriefcaseManager.downloadNewBriefcaseAndOpen
+      return briefcaseDb;
     }
 
-    /*
-     * Open the briefcase
-     * Note: This call must be made even if the briefcase is already open - this is to ensure the usage is logged
-     */
-    const briefcaseDb: BriefcaseDb = await BriefcaseDb.open(requestContext, briefcaseProps.key);
-    Logger.logTrace(loggerCategory, "RpcBriefcaseUtility.openWithTimeout: Opened briefcase", () => ({ ...tokenProps, syncMode }));
-    return briefcaseDb.getConnectionProps();
+    const checkpoint: CheckpointProps = {
+      iModelId: tokenProps.iModelId!,
+      contextId: tokenProps.contextId!,
+      changeSetId: tokenProps.changeSetId!,
+      requestContext,
+    };
+
+    // opening a checkpoint, readonly.
+    let db: SnapshotDb | void;
+    try {

```

Comment:
Should we skip this entirely if `process.env.BLOCKCACHE_DIR` is not set?

Reply:
I don't think that's worth it.